Remove incorrect close

NickeZ opened the pull request:
Ownership of the slave file descriptor is passed to the process::Stdio
structs and will be closed when they are dropped. If the slave fd is
closed explicitly then the next fd that is opened during the same scope
will get silently closed by the Stdio structs when they are dropped.

I had issues with race conditions in another software project so I think it affects this project to. Luckily no fds are opened between `close(slave)` and the end of that scope.

Do you need more details for accepting this?

https://doc.rust-lang.org/std/os/unix/io/trait.FromRawFd.html#tymethod.from_raw_fd

NickeZ commented:
Should be closed here:

https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/fd.rs#L245

As far as I understand the internals of rust stdlib..

jwilm commented:
Makes sense. I was thinking about the builder being "moved" to the child so still needing to be closed manually, but that's not how fork works :upside_down_face:.

Thanks for the fix! :heart: code deletion patches.

NickeZ commented:
Here you have a small test as well =) Notice that `/dev/pts` is missing in the second listing.

```diff
iff --git a/src/main.rs b/src/main.rs
index 5126b09..c3f8107 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -104,6 +104,7 @@ fn run(mut config: Config, options: cli::Options) -> Result<(), Box<Error>> {
     // pseudoterminal. A file descriptor for the master side is retained for
     // reading/writing to the shell.
     let mut pty = tty::new(&config, &options, display.size());
+    tty::printfds("dropped");
     // Create the pseudoterminal I/O loop
     //
diff --git a/src/tty.rs b/src/tty.rs
index bd291b1..fd69ddd 100644
--- a/src/tty.rs
+++ b/src/tty.rs
@@ -63,6 +63,16 @@ fn errno() -> c_int {
     ::errno::errno().0
 }
+pub fn printfds(prefix: &str) {
+    for entry in ::std::fs::read_dir("/proc/self/fd").unwrap() {
+        let entry = entry.unwrap();
+        if let Ok(canon_path) = ::std::fs::canonicalize(entry.path()) {
+            let path = entry.path();
+            debug!("{}: {:?} {:?}", prefix, path, canon_path);
+        }
+    }
+}
+
 /// Get raw fds for master/slave ends of a new pty
 #[cfg(target_os = "linux")]
 fn openpty(rows: u8, cols: u8) -> (c_int, c_int) {
@@ -181,6 +191,8 @@ pub fn new<T: ToWinsize>(config: &Config, options: &Options, size: T) -> Pty {
     let (master, slave) = openpty(win.ws_row as _, win.ws_col as _);
+    printfds("created");
+
     let default_shell = &Shell::new(pw.shell);
     let shell = config.shell()
         .unwrap_or(&default_shell);
```

```
 ./target/debug/alacritty -vv -e bash
Welcome to Alacritty.
device_pixel_ratio: 1
width: 1024, height: 768
Initializing glyph cache
Finished initializing glyph cache in 0.067738777
set_inner_size: 640px × 360px
width: 644, height: 644
Cell Size: (8 x 15)
PTY Dimensions: Line(42) x Column(80)
created: "/proc/self/fd/0" "/dev/pts/1"
created: "/proc/self/fd/1" "/dev/pts/1"
created: "/proc/self/fd/2" "/dev/pts/1"
created: "/proc/self/fd/4" "/dev/dri/card1"
created: "/proc/self/fd/5" "/dev/ptmx"
created: "/proc/self/fd/6" "/dev/pts/20"
created: "/proc/self/fd/7" "/proc/11159/fd"
dropped: "/proc/self/fd/0" "/dev/pts/1"
dropped: "/proc/self/fd/1" "/dev/pts/1"
dropped: "/proc/self/fd/2" "/dev/pts/1"
dropped: "/proc/self/fd/4" "/dev/dri/card1"
dropped: "/proc/self/fd/5" "/dev/ptmx"
dropped: "/proc/self/fd/6" "/proc/11159/fd"
Term::resize
```
